docs/custom-domain/registry-setup.md

```diff
@@ -0,0 +1,94 @@
+  
+- ACR Managed Identities:\
+  Managed Identities (MI) provide a mechanism to associate an Azure Active Directory identity with your registry, while relieving you of the burden of managing credentials. To learn more, see the documentation [here](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview).\
+ ACR supports both user assigned and system assigned MI. If you choose to enable user assigned MI for your registry, create one first before proceeding. Instructions [here](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal).
+
+### Enable preview features
+1. `az login`
+2. `az account set -s <subscription-id-or-name> `
+3. `apiVersion="api-version=2019-12-01-preview"`
+4. `mgmtEndpoint=$(az cloud show --query endpoints.resourceManager -o tsv)`
+5. `accessToken=$(az account get-access-token --query accessToken -o tsv)`
+6. `registryResourceId=$(az acr show -n myregistry --query id -o tsv)`
+7. You can either enable a system assigned MI, or a user assigned MI, or both for your registry. We recommend using a system assigned MI to enable advanced scenarios with virtual networks that are [called out below](#enhanced-security-with-virtual-networks). Do one of the following:
+   - System assigned: 
+     - `identity="{\"type\": \"systemAssigned\"}"`
+   - User assigned: 
+     - `userMIResourceId=$(az identity show -n <user-mi-name> -g <user-mi-resource-group> --subscription <user-mi-subscription-id-or-name> --query id -o tsv)`
+     - `identity="{\"type\": \"userAssigned\", \"userAssignedIdentities\": {\"${userMIResourceId}\":{}}}"`
+   - Both:
+     - `userMIResourceId=$(az identity show -n <user-mi-name> -g <user-mi-resource-group> --subscription <user-mi-subscription-id-or-name> --query id -o tsv)`
+     - `identity="{\"type\": \"systemAssigned,userAssigned\", \"userAssignedIdentities\": {\"${userMIResourceId}\":{}}}"`
+8. `payload="{\"identity\": $identity, \"properties\": {\"dataEndpointEnabled\": true}}"`
+9. `endpoint="$mgmtEndpoint$registryResourceId?$apiVersion"`
+10. `curl -f -X PATCH -H "Authorization: Bearer $accessToken" -H "Content-Type: application/json" $endpoint -d "$payload"`
+
+## Prepare your Azure Key Vault
+For each domain, its TLS private key and public certificate pair must be added to an Azure Key Vault that is accessible by your registry as a single PEM formatted file. We recommend creating a new key vault containing only your TLS certificates and granting the registry's identity access to `get` secret.
+1. [Create](https://docs.microsoft.com/azure/key-vault/) a new Azure Key Vault.
+2. [Add](https://docs.microsoft.com/azure/key-vault/certificate-scenarios) your certificates to the key vault.
+3. Add an access policy to the key vault that grants your registry's identity access to `get` secret:\
+   `az keyvault set-policy --name <your-kv-name> --secret-permissions get --spn <registry-system-or-user-mi-client-id>`
```

Comment:
Do you provide a command to obtain the id of the system-assigned managed identity?

Reply:
Okay, added some commands